core_backend/tests/api/test_data_api.py

```diff
@@ -37,6 +43,145 @@ def utcnow(self) -> datetime:
         return self.date
 
 
+class TestUrgencyQueryDataAPI:
+    @pytest.fixture
+    async def user1_data(
+        self,
+        monkeypatch: pytest.MonkeyPatch,
+        asession: AsyncSession,
+        urgency_rules: int,
+    ) -> AsyncGenerator[None, None]:
+        dates = [
+            datetime.utcnow() - relativedelta(days=x) for x in range(N_DAYS_HISTORY)
+        ]
+        all_orm_objects: List[Any] = []
+
+        for i, date in enumerate(dates):
+            monkeypatch.setattr(
+                "core_backend.app.urgency_detection.models.datetime", MockDatetime(date)
+            )
+            urgency_query = UrgencyQuery(message_text=f"query {i}")
+            urgency_query_db = await save_urgency_query_to_db(
+                1, "secret_key", urgency_query, asession
+            )
+            all_orm_objects.append(urgency_query_db)
+            is_urgent = i % 2 == 0
+
+            urgency_response = UrgencyResponse(
+                is_urgent=is_urgent,
+                matched_rules=["rule1", "rule2"],
+                details={
+                    1: UrgencyRuleCosineDistance(urgency_rule="rule1", distance=0.4)
+                },
+            )
+            urgency_response_db = await save_urgency_response_to_db(
+                urgency_query_db, urgency_response, asession
+            )
+            all_orm_objects.append(urgency_response_db)
+
+        yield
+
+        for orm_object in reversed(all_orm_objects):
+            await asession.delete(orm_object)
+        await asession.commit()
+
+    @pytest.fixture
+    async def user2_data(
+        self,
+        monkeypatch: pytest.MonkeyPatch,
+        asession: AsyncSession,
+    ) -> AsyncGenerator[int, None]:
+
+        days_ago = random.randrange(N_DAYS_HISTORY)
+        date = datetime.utcnow() - relativedelta(days=days_ago)
+        monkeypatch.setattr(
+            "core_backend.app.urgency_detection.models.datetime", MockDatetime(date)
+        )
+        urgency_query = UrgencyQuery(message_text="query")
+        urgency_query_db = await save_urgency_query_to_db(
+            2, "secret_key", urgency_query, asession
+        )
+        yield days_ago
+        await asession.delete(urgency_query_db)
+        await asession.commit()
+
+    def test_urgency_query_data_api(
+        self,
+        user1_data: pytest.FixtureRequest,
+        client: TestClient,
+    ) -> None:
+
+        response = client.get(
+            "/data-api/urgency-queries",
+            headers={"Authorization": "Bearer test_api_key"},
+            params={"start_date": "2021-01-01T00:00", "end_date": "2021-01-10T00:00"},
+        )
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize(
+        "start_day, end_day", [[5, 1], [6, 4], [5, 5], [0, 0], [20, 14], [11, 0]]
```

Comment:
Also, it seems like we are allowing start_date > end_date in query params (just returns empty list. If this is intentional should we add such test cases too? e.g. [1, 5], [4, 6], ...

Reply:
Good idea. I'll add that